tests/unit/amo/components/TestSearchForm.js

```diff
@@ -1,179 +1,386 @@
+      wrapper = mountComponent({
+        query: 'foo',
+        dispatch,
+      });
+      // no call to handleSuggestionsFetchRequested() until the input has focus,
+      // even if there is already a `searchValue`
+      expect(dispatch.calledOnce).toBe(false);
+      // this is needed to trigger handleSuggestionsFetchRequested()
+      wrapper.find('input').simulate('focus');
+      expect(dispatch.calledOnce).toBe(true);
+      expect(dispatch.calledWith(autocompleteStart({
+        errorHandlerId: 'error-handler-id',
+        filters: {
+          query: 'foo',
+        },
+      }))).toBe(true);
+    });
+
+    it('clears suggestions when input is cleared', () => {
+      const dispatch = sinon.spy();
+      wrapper = mountComponent({
+        query: 'foo',
+        dispatch,
+      });
+      // clearing the input calls handleSuggestionsClearRequested()
+      wrapper.find('input').simulate('change', createFakeChangeEvent());
+      expect(dispatch.calledOnce).toBe(true);
+      expect(dispatch.calledWith(autocompleteCancel())).toBe(true);
+    });
+
+    it('displays suggestions when user is typing', () => {
+      const { store } = dispatchAutocompleteResults({ results: [
+        createFakeAutocompleteResult(),
+        createFakeAutocompleteResult(),
+      ] });
+      const { autocomplete: autocompleteState } = store.getState();
 
-  it('passes the context through', () => {
-    const { store } = dispatchSignInActions();
-    store.dispatch(setViewContext(ADDON_TYPE_EXTENSION));
+      wrapper = mountComponent({
+        query: 'foo',
+        suggestions: autocompleteState.suggestions,
+      });
+      expect(wrapper.find(Suggestion)).toHaveLength(0);
+      // this triggers Autosuggest
+      wrapper.find('input').simulate('focus');
+      expect(wrapper.find(Suggestion)).toHaveLength(2);
+      expect(wrapper.find(LoadingText)).toHaveLength(0);
+    });
+
+    it('does not display suggestions when search is empty', () => {
+      const { store } = dispatchAutocompleteResults({ results: [
+        createFakeAutocompleteResult(),
+        createFakeAutocompleteResult(),
+      ] });
+      const { autocomplete: autocompleteState } = store.getState();
+
+      // setting the `query` prop to empty also sets the input state to empty.
+      wrapper = mountComponent({
+        query: '',
+        suggestions: autocompleteState.suggestions,
+      });
+      wrapper.find('input').simulate('focus');
+      expect(wrapper.find(Suggestion)).toHaveLength(0);
+    });
+
+    it('does not display suggestions when there is no suggestion', () => {
+      wrapper = mountComponent({ suggestions: [] });
+      wrapper.find('input').simulate('focus');
+      expect(wrapper.find(Suggestion)).toHaveLength(0);
+      expect(wrapper.find(LoadingText)).toHaveLength(0);
+    });
+
+    it('displays 10 loading bars when suggestions are loading', () => {
+      wrapper = mountComponent({
+        suggestions: [],
+        loadingSuggestions: true,
```

Comment:
Instead of setting props explicitly, this should dispatch the `autocompleteStart` action and do something like:

```js
wrapper = mountComponent(mapStateToProps(store.getState()));
```